### ISSUE TITLE
Problem on line 604 Armorsmith

### ISSUE THREAD

**Issue by twgamers** [samclane/Snake-Cogs#8]
File "C:\Users\PC\Desktop\RED\Red-DiscordBot\cogs\armorsmith.py", line 604, in challenge
    battle_text, author_won = self.duel(author, user, settings)
TypeError: 'coroutine' object is not iterable

**Comment by samclane** [comment]
I took a look, and I think I found the issue. I pushed the new changes (c85aeeabe22671dfd664512923fe744dcc3b99c1). Let me know if they worked.

**Comment by dumptruckman** [comment]
Your changes did not fix the issue.
```
Exception in command 'fight challenge'
Traceback (most recent call last):
  File "lib/discord/ext/commands/core.py", line 50, in wrapped
    ret = yield from coro(*args, **kwargs)
  File "/home/mvbotuser/Red-DiscordBot/cogs/armorsmith.py", line 604, in challenge
    battle_text, author_won = self.duel(author, user, settings)
TypeError: 'coroutine' object is not iterable

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "lib/discord/ext/commands/bot.py", line 846, in process_commands
    yield from command.invoke(ctx)
  File "lib/discord/ext/commands/core.py", line 634, in invoke
    yield from ctx.invoked_subcommand.invoke(ctx)
  File "lib/discord/ext/commands/core.py", line 374, in invoke
    yield from injected(*ctx.args, **ctx.kwargs)
  File "lib/discord/ext/commands/core.py", line 54, in wrapped
    raise CommandInvokeError(e) from e
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: TypeError: 'coroutine' object is not iterable```

**Comment by dumptruckman** [comment]
I think I fixed it in #9
